lib/ansible/plugins/inventory/foreman.py

```diff
@@ -270,6 +294,18 @@ def _populate(self):
                             self.display.warning("Could not set hostvar ansible_ssh_host to '%s' for the '%s' host, skipping: %s" %
                                                  (host['ip'], host_name, to_text(e)))
 
+                if self.get_option('group_patterns'):
```

Comment:
might want to check if `self.get_option('keyed_groups')` is non-null, and if it is, might be best to throw an error or a warning. I'm not sure exactly the precedent for handling this.